### DISCUSSION TITLE
Can I replace a specific cell (row:2 , column2 etc) with a custom component?

### DISCUSSION THREAD

**Discussion by chairedevil** [nick-keller/react-datasheet-grid#67] [Q&A] [answered]
I'm very interested in your datasheet.
but in my project. I need to replace a specific cell (row:2, column2) with a custom component.
Because each row has a different type of data to display.
Can I accomplish this?

**Comment by chairedevil** [comment, marked as answer]
I handle with this way.

`{ ...keyColumn('test', textColumn), title: 'TEST', component: TestInput },`

```
const TestInput = (data: any) => {
  let input
  if (data.rowData.type === 0) {
    input = <input type="text" value={data.rowData.test} />
  } else {
    input = <input type="text" value={data.rowData.test} style={{ backgroundColor: 'red' }} />
  }
  return data.focus ? input : <div>{data.rowData.test}</div>
}
```

**Comment by nick-keller** [comment]
Hello @chairedevil ,
Yes you can, you would need to add conditions in the component, just like you did.
Note that if this is just a matter of style, you can use [cellClassName](https://react-datasheet-grid.netlify.app/docs/api-reference/columns#cellclassname).

I hope that answers your question, feel free to elaborate if you have any further questions ;)

**Reply by chairedevil** [reply]
thankyou so much
